.gitignore

```diff
@@ -1,42 +1,176 @@
-dist
-*classes
-target/
-build/
-.gradle/
+# Compiled from https://github.com/github/gitignore
+# With Kafka Specific alterations
+
+# Covers Linux OS
+*~
+## KDE directory preferences
+.directory
+## Linux trash folder which might appear on any partition or disk
+.Trash-*
+
+# Covers Windows OS
+## Windows image file caches
+Thumbs.db
+ehthumbs.db
+## Folder config file
+Desktop.ini
+## Recycle Bin used on file shares
+$RECYCLE.BIN/
+## Windows Installer files
```

Comment:
Why would installer files ever be in a Kafka code checkout? I think stuff like this we can trim. The "global" gitignore configs that make sense to include are ones that we have reason to believe would show up commonly in a developer's checkout and aren't covered by other rules (e.g. even if we one day produced a windows installer, it'd end up in `target/` anyway). I'm thinking things like the thumbnail caches, editor backup files, etc.